RebuyFormsRating.Droid/FeedbackSheet.cs

```diff
@@ -86,16 +85,32 @@ void bv_click(object sender, EventArgs e)
         {
             var bv = sender as Android.Widget.Button;
             if (bv != null) {
+                var feedback = new RatingViewFeedback();
                 var s = bv.Text;
                 if (String.Equals(s, strSend)) {
                     var tv = (EditText)customFeedbackSheet?.FindViewById(Resource.Id.feedbackText);
+                    var mv = (EditText)customFeedbackSheet?.FindViewById(Resource.Id.feedbackMail);
+
+                    if (!string.IsNullOrEmpty(mv.EditableText.ToString()) && !isValidEmail(mv.EditableText.ToString())) {
+                        mv.SetError("Ungültige E-Mail-Adresse", null);
+
+                        return;
+                    }
 
-                    tcs.TrySetResult(tv?.EditableText.ToString());
-                } else {
-                    tcs.TrySetResult(string.Empty);
+                    feedback.Feedback = tv.EditableText.ToString();
+                    feedback.Email = mv.EditableText.ToString();
                 }
-                dialog.Dismiss();
+                tcs.TrySetResult(feedback);
             }
+            dialog.Dismiss();
+        }
+
+        bool isValidEmail(string target)
+        {
+            if (target == null)
```

Comment:
you could as well use here a normal one-line solution

`return  target != null && ..... Matcher(target).Matches()`